### PR TITLE
fix logic for testmode

### PR DESCRIPTION
**Pull request by deathbearbrown**
I am also tired of me breaking everything :sob:

### CODE REVIEW

**Comment by bmac** [comment]
+1

**Comment by deathbearbrown** [comment]
Actually -1, it's breaking tests I think

**Comment by deathbearbrown** [comment]
@bmac @carldanley @jugglinmike please review :)

**Comment by jugglinmike** [comment]
I can't speak for the change to `BaseEndpoint.php` (still learning the ropes for ae-api), but everything else looks good to me.

**Comment by jugglinmike** [comment]
(and the resolution of gh-13 should make this a bit easier)

File: utils/database.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ public static function connect() {
 
     //throw errors, but not too many
     mysqli_report(MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT);
-    if (defined('TESTMODE')) {
+    if (TESTMODE === 'true') {
```
Comment (bmac) [inline utils/database.php]: What happens if `TESTMODE` is `true` is `===` a non strict comparison in php?

**Comment by deathbearbrown** [comment]
the Nginx config that would be needed to make this work removes query params >.<

**Comment by deathbearbrown** [comment]
I fixed it, I think and I might cry because I did it all by myself ;_; 
```
server {
  listen        80;
  server_name   api.dev;
  root          /mnt/src;

  location / {
    index index.php index.html;
    try_files $uri $uri/ /index.php?uri=$request_uri&query=$args;
  }

  location ~* /test(.*)$ {
   try_files $uri $uri/ /index.php?testmode=true&uri=$1&query=$args;
  }

  # Directives to send expires headers and turn off 404 error logging.
  location ~* \.(js|css|png|jpg|jpeg|gif|ico)$ {
    expires 24h;
    log_not_found off;
  }

  # this prevents hidden files (beginning with a period) from being served
  location ~ /\. {
    access_log off;
    log_not_found off;
    deny all;
  }

  location ~ \.php$ {
    client_max_body_size 25M;
    try_files $uri =404;

    fastcgi_split_path_info ^(.+\.php)(/.+)$;

    include fastcgi_params;
    fastcgi_index index.php;
    fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
    fastcgi_pass php;
    fastcgi_param TESTMODE $arg_testmode;
    fastcgi_param REQUEST_URI $arg_uri;
    fastcgi_param QUERY_STRING $arg_query;
  }

  error_log     /mnt/logs/error.log;
  access_log    /mnt/logs/access.log;
}
```

**Comment by bmac** [comment]
Yay @deathbearbrown \o/. You are the best!

File: utils/database.php (hunk 1 above)
Comment (carldanley) [inline utils/database.php]: Maybe just `if (TESTMODE) {` ?

**Comment by carldanley** [comment]
Awesome work @deathbearbrown, reviewing code now.

**Comment by carldanley** [comment]
code :+1: tests :+1:

File: utils/database.php (hunk 1 above)
Comment (deathbearbrown) [inline utils/database.php]: both of these suggestions break btw

**Comment by deathbearbrown** [comment]
if this is good to merge then I can add a new tag and do versioning edits to the other 2 prs

**Comment by MattSurabian** [comment]
I think @jugglinmike is currently reviewing as well, might want to wait

**Comment by deathbearbrown** [comment]
we good?

**Pull request merged**